tests/jballerina-unit-test/src/test/resources/test-src/query/query-exp-iterable-objects.bal

```diff
@@ -196,3 +197,32 @@ class IterableFromIterator {
 function getIterableObject(_Iterator iterator) returns IterableFromIterator {
     return new IterableFromIterator(iterator);
 }
+
+class MyIterable {
+    *object:Iterable;
+    public function iterator() returns object {
+                                           public function next() returns record {| int value; |}?;
+                                       } {
+        return new MyIterator();
+    }
+}
+
+int i = 0;
+
+public class MyIterator {
+    public function next() returns record {| int value; |}? {
+        i += 1;
+        if (i < 5) {
+            return {value: i};
+        }
+        return ();
+    }
+}
+
+public function testObjectIterator() {
+    int[] expectedArr = [1, 2, 3, 4];
+    int[] integers = from var item in new MyIterable()
+                     select item;
+
+    test:assertEquals(integers, expectedArr);
```

Comment:
Shall we write a ballerina function to panic on mismatch instead of importing `ballerina/test`? Ideally the lang tests shouldn't import non-lang modules.

Reply:
Updated in ae27020eac86554811519ffe8440e93416aeb22a